scripts/calibrateArmDac.sh

```diff
@@ -1,54 +1,74 @@
 #!/bin/bash
-# Usage:
-#   calibrateArmDac.sh <DetName> <cardName> <link> <vfatmask> <Comma Separated List of CFG_THR_ARM_DAC Values>
+usage() {
+    echo './calibrateArmDac.sh -s slot -l link  -L armdaclist [-S shelf] [-m vfatmask] [-r path]'
+    echo ''
+    echo '    armdaclist: comma separate list of CFG_THR_ARM_DAC Values (e.g. 25,30,35,40,45,50)'
+    echo '    link: OH number on the CTP7'
+    echo '    shelf: Shelf number'
+    echo '    slot: Slot number'
+    echo '    vfatmask: 24-bit mask where a 1 in the n^th bit implies the n^th VFAT shall be excluded'
+    echo '    path: output directory to be used instead of creating a new scandated directory'
+    echo ''
+    echo 'For each CFG_THR_ARM_DAC value specified will create a scandate directory, configure, print the configuration, and take an scurve. If "-r path" is provided, ARM DAC values for which an scurve log file already exists in the provided directory will be skipped.'
+    kill -INT $$;
+}
 
-helpstring="Usage:
-    ./calibrateArmDac.sh [DetName] [cardName] [shelf]  [slot] [link] [vfatmask] [comma separated list of CFG_THR_ARM_DAC Values]
-    ./calibrateArmDac.sh -r [PATH] [DetName] [cardName] [shelf] [slot] [link] [vfatmask] [comma separated list of CFG_THR_ARM_DAC Values]
-        
-    For each CFG_THR_ARM_DAC value specified will create a scandate directory, configure, print the configuration, and take an scurve
 
-        DetName: Serial Number of Detector, this should be a subfolder under DATA_PATH variable
-        cardName: CTP7 network alias or ip address
-        link: OH number on the CTP7
-        shelf: Shelf number
-        slot: Slot number 
-        vfatmask: 24-bit mask where a 1 in the n^th bit implies the n^th VFAT shall be excluded
-
-    If -r [PATH] is provided, the directory [PATH] will be used instead of creating a new scandate directory and ARM DAC values for which a scurve log file already exists in that directory will be skipped
-"
 
 ISFILE=0;
 OPTIND=1
-while getopts "r:h" opts
+while getopts "r:S:s:l:m:L:h" opts
 do
     case $opts in
         r)
             RESUMEDIR=$OPTARG;;
+        S)
+            SHELF=$OPTARG;;
+        s)
+            SLOT=$OPTARG;;
+        l)
+            LINK=$OPTARG;;
+        m)
+            MASK=$OPTARG;;
+        L)
+            LIST_ARM_DAC=$OPTARG;;
```

Comment:
my ignorance here.  But if the user doesn't supply these they won't exist right?  Does that cause problems later on when trying to apply them?  Particularly in the case of `MASK`?

Reply:
The script exits if any of `LIST_ARM_DAC`, `SLOT`, or `LINK` is not set. For `DETECTOR`, `SHELF`,  and `MASK` it will set them if they are not set. `RESUME_DIR`, if it exists, is used to set `OUTDIR` and then is not used later.